cg/meta/workflow/analysis.py

```diff
@@ -0,0 +1,341 @@
+            out_dir=self.get_trailblazer_config_path(case_id=case_id).parent.as_posix(),
+            config_path=self.get_trailblazer_config_path(case_id=case_id).as_posix(),
+            priority=self.get_priority_for_case(case_id=case_id),
+            data_analysis=str(self.pipeline),
+        )
+
+    def get_hermes_transformed_deliverables(self, case_id: str) -> dict:
+        return self.hermes_api.create_housekeeper_bundle(
+            bundle_name=case_id,
+            deliverables=self.get_deliverables_file_path(case_id=case_id),
+            pipeline=str(self.pipeline),
+            analysis_type=self.get_bundle_deliverables_type(case_id),
+            created=self.get_bundle_created_date(case_id),
+        ).dict()
+
+    def get_bundle_created_date(self, case_id: str) -> dt.datetime:
+        raise NotImplementedError
+
+    def get_pipeline_version(self, case_id: str) -> str:
+        raise NotImplementedError
+
+    def set_statusdb_action(self, case_id: str, action: Optional[str]) -> None:
+        """
+        Set one of the allowed actions on a case in StatusDB.
+        """
+        if action in [None, *CASE_ACTIONS]:
+            case_obj: models.Family = self.status_db.family(case_id)
+            case_obj.action = action
+            self.status_db.commit()
+            LOG.info("Action %s set for case %s", action, case_id)
+            return
+        LOG.warning(
+            f"Action '{action}' not permitted by StatusDB and will not be set for case {case_id}"
+        )
+
+    def get_analyses_to_clean(self, before: dt.datetime) -> List[models.Analysis]:
+        analyses_to_clean = self.status_db.analyses_to_clean(pipeline=self.pipeline, before=before)
+        return analyses_to_clean.all()
+
+    def get_cases_to_analyze(self) -> List[models.Family]:
+        return self.status_db.cases_to_analyze(
+            pipeline=self.pipeline, threshold=self.threshold_reads
+        )
+
+    def get_running_cases(self) -> List[models.Family]:
+        return (
+            self.status_db.query(models.Family)
+            .filter(models.Family.action == "running")
+            .filter(models.Family.data_analysis == self.pipeline)
+            .all()
+        )
+
+    def get_cases_to_store(self) -> List[models.Family]:
+        raise NotImplementedError
+
+    def get_sample_fastq_destination_dir(self, case_obj: models.Family, sample_obj: models.Sample):
+        raise NotImplementedError
+
+    def gather_file_metadata_for_sample(self, sample_obj: models.Sample) -> List[dict]:
+        return [
+            self.fastq_handler.parse_file_data(file_obj.full_path)
+            for file_obj in self.housekeeper_api.files(
+                bundle=sample_obj.internal_id, tags=["fastq"]
+            )
+        ]
+
+    def link_fastq_files_for_sample(
+        self, case_obj: models.Family, sample_obj: models.Sample, concatenate: bool = False
+    ) -> None:
+        """Link FASTQ files for a sample."""
```

Comment:
Looks like this function does a lot more. Could you describe that in the docstring?

Reply:
done